linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -1455,10 +1455,8 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
       case NothingType =>
         cont(treceiver)
       case NullType =>
-        cont(Block(
-            finishTransformStat(treceiver),
-            Throw(New(NullPointerExceptionClass,
-                MethodIdent(NoArgConstructorName), Nil))).toPreTransform)
+        // Apply on null is UB, just create a well-typed tree.
+        cont(Throw(Null()).toPreTransform)
```

Comment:
We must not throw away the `treceiver`. It could be that `treceiver` itself throws, in which case the behavior is well-defined: that thrown exception must be kept.

So this should be something like a `Block(finishTransformStat(treceiver), Throw(Null()))`.

Reply:
Hah :) I knew there was a reason why this is too much :P